src/main.rs

```diff
@@ -290,9 +290,13 @@ async fn main_impl() -> Result<()> {
             let Ticket {
                 hash,
                 peer,
-                addr,
+                addrs,
                 token,
             } = ticket;
+            let addr = addrs
+                .get(0)
+                .copied()
+                .context("missing SocketAddr in ticket")?;
```

Comment:
this is bascially ignoring this problem for now in our own implementation.  i don't feel like spending time on this yet.  I'll come back to this once working on trying to connect to all addresses at the same time.

Reply:
please file an issue, this will break easily. in most setups if we don’t try all addrs